### PR TITLE
feat: Moderation Console

### PR DESCRIPTION
**Pull request by Dnouv**
### Why this Moderation Console and what's this?
In the past, before this Moderation Console, any report against a message was just stored in the DB, and there was no way to view it or take appropriate action against them to moderate these bad/harmful users.

And to the rescue, the "Moderation console" provides the Moderator a way to view & manage the messages or users that were reported across different channels.

On the Moderation console, we have each row a User (only users whose messages were reported), and when you click on the row, you will see the Reported Messages. And also a set of actions that you could take straight from here, which includes deleting reported messages, deactivating the user, or resetting the profile picture (for the users reported due to some strange profile pictures.)

The PR works on implementing the client-side Moderation Console. Here is the preview of the current UI:

Old:
image

Updated:

Screenshot 2023-03-30 011233
Screenshot 2023-03-30 011819
Screenshot 2023-03-30 011737
Screenshot 2023-03-30 011723

### CODE REVIEW

File: apps/meteor/client/views/admin/moderation/ModerationConsoleTable.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const {
+		data,
+		refetch: reloadReports,
+		isLoading,
+		isSuccess,
+	} = useQuery(
+		['reports', query],
```
Comment (hugocostadev) [inline apps/meteor/client/views/admin/moderation/ModerationConsoleTable.tsx:63]: same as the other key

File: apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json [hunk 2]
```diff
@@ -631,12 +631,17 @@
   "is_uploading": "is uploading",
   "is_recording": "is recording",
   "Are_you_sure": "Are you sure?",
+  "Are_you_sure_dismiss_and_delete_reports": "Are you sure you want to dismiss and delete all reports for this user's messages? This action cannot be undone.",
```
[the remark this answers is not in the document]
Comment (Dnouv) [reply apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json]: Thank you! Fixed.

File: apps/meteor/client/views/admin/moderation/UserMessages.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const {
+		data: userMessages,
+		refetch: reloadUserMessages,
+		isLoading: isLoadingUserMessages,
+		isSuccess: isSuccessUserMessages,
+	} = useQuery(
+		['userMessages', query],
```
[the remark this answers is not in the document]
Comment (Dnouv) [reply apps/meteor/client/views/admin/moderation/UserMessages.tsx]: Thank you! Fixed.

File: apps/meteor/client/views/admin/moderation/MessageReportInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	const formatTime = useFormatTime();
+	const formatDate = useFormatDate();
+
+	const {
+		data: reportsByMessage,
+		isLoading: isLoadingReportsByMessage,
+		isSuccess: isSuccessReportsByMessage,
+		isError: isErrorReportsByMessage,
+	} = useQuery(
+		['reportsByMessage', { msgId }],
```
[the remark this answers is not in the document]
Comment (Dnouv) [reply apps/meteor/client/views/admin/moderation/MessageReportInfo.tsx]: Thank you! Fixed.

File: apps/meteor/client/views/admin/moderation/ModerationConsoleTable.tsx (hunk 1 above)
Comment (Dnouv) [reply apps/meteor/client/views/admin/moderation/ModerationConsoleTable.tsx]: Thank you! Fixed.

**Review by hugocostadev — APPROVED** [review]

**Pull request merged**